include/swift/AST/RawComment.h

```diff
@@ -78,6 +78,19 @@ struct CommentInfo {
   uint32_t SourceOrder;
 };
 
+struct LineColumn {
+  uint32_t Line;
+  uint32_t Column;
+};
+
+struct BasicDeclLocs {
+  StringRef SourceFilePath;
+  Optional<LineColumn> Loc;
+  Optional<LineColumn> NameLoc;
```

Comment:
I don't think we have any decls where Loc and NameLoc are different. Is this just to account for decls without names?

Reply:
We have separate APIs `getLoc()` and `getNameLoc()` for `ValueDecl`, so they could potentially be different. IMO, we shouldn't hardcode that they are usually identical to the format.

Reply:
I'll flip it around: there are zero cases where `getLocFromSource()` does not return `getNameLoc()` if there is a `getNameLoc()`, and we could put that in the AST verifier.